packages/router/src/util.ts

```diff
@@ -622,18 +620,23 @@ export function analyzeRoutes(
         }
 
         if (children) {
+          const newSetId =
```

Comment:
I like this solution thoughts on extracting it out into its own util?
If we extract it makes it a bit clearer what the conditional is actually doing also easily testable

```
{
   id: createSetId(previousSets, nextSetId)
}
```

Reply:
Also side note this could be potentially useful with nested routes

Reply:
I like this! I extracted a `createSetId` function and added some comments and stuff to make it easier to read/understand.